addons/mail/views/mail_template_views.xml

```diff
@@ -8,51 +8,35 @@
                 <form string="Templates">
                     <header>
                         <field name="template_fs" invisible="1"/>
+                        <field name="current_user_is_template_admin" invisible="1"/>
+                        <button string="Preview"
+                                name="%(mail_template_preview_action)d" type="action"
+                                class="btn-primary" target="new"/>
                         <button string="Reset Template"
                                 name="%(mail_template_reset_action)d" type="action"
                                 groups="mail.group_mail_template_editor"
                                 attrs="{'invisible': [('template_fs', '=', False)]}"/>
+                        <button string="Delete"
+                                name="unlink" type="object"
+                                class="btn-secondary"
+                                attrs="{'invisible': &quot;[
+                                    '|',
+                                        ('template_fs', '!=', False),
+                                    ('current_user_is_template_admin', '=', False),
+                                    ('user_id', '!=', uid)
+                                ]&quot;}"
```

Comment:
It was necessary in this case - something to do with uid not being evaluated by the JS, so you have to send a string evaluated by the server instead

Reply:
Ah yes I remember that you talked about it some time ago, did you touch a word with the framework teams to see what would be the cause ?

Reply:
I did not, I'm guessing there's an explanation for it but since invisible is used that way in around 20 other locations in the code, I'm guessing this is how it's supposed to be used with uid